### PR TITLE
Fixing background work invocations

### PR DESCRIPTION
**Pull request by sithhell**
flyby: fix inspect

### CODE REVIEW

File: src/runtime/threads/detail/thread_pool.cpp [hunk 1]
```diff
@@ -43,6 +44,14 @@
 
 namespace hpx { namespace threads { namespace detail
 {
+    thread_id_type& background_thread()
+    {
+        struct tls_tag {};
+        static hpx::util::thread_specific_ptr<thread_id_type, tls_tag> thrd;
+        if (thrd.get() == nullptr) thrd.reset(new thread_id_type());
```
Comment (hkaiser) [inline src/runtime/threads/detail/thread_pool.cpp]: Where is this deleted?

File: src/runtime/threads/detail/thread_pool.cpp [hunk 2]
```diff
@@ -43,6 +44,14 @@
 
 namespace hpx { namespace threads { namespace detail
 {
+    thread_id_type& background_thread()
+    {
+        struct tls_tag {};
+        static hpx::util::thread_specific_ptr<thread_id_type, tls_tag> thrd;
```
Comment (hkaiser) [inline src/runtime/threads/detail/thread_pool.cpp]: Why is the background thread stored in a thread-local variable. Wouldn't be sufficient to have it on the stack as before?

File: hpx/runtime/threads/detail/scheduling_loop.hpp [hunk 3]
```diff
@@ -287,6 +287,69 @@ namespace hpx { namespace threads { namespace detail
...
+                {
+                    // tries to set state to active (only if state is still
+                    // the same as 'state')
+                    detail::switch_status thrd_stat (background_thread().get(), state);
+                    if (HPX_LIKELY(thrd_stat.is_valid() &&
+                            thrd_stat.get_previous() == pending))
+                    {
+                        thrd_stat = (*background_thread())();
+
+                        thread_data *tmp = thrd_stat.get_next_thread();
```
Comment (hkaiser) [inline hpx/runtime/threads/detail/scheduling_loop.hpp]: Could we use a variable name which is more descriptive, please?

File: hpx/runtime/threads/detail/scheduling_loop.hpp [hunk 4]
```diff
@@ -287,6 +287,69 @@ namespace hpx { namespace threads { namespace detail
...
+    // This function tries to invoke the background work thread. It returns
+    // false when we need to directly invoke the background thread function, true
+    // otherwise
+    template <typename SchedulingPolicy>
+    bool call_background_thread(
+        thread_data*& next_thrd,
+        SchedulingPolicy& scheduler,
+        std::size_t num_thread
+    )
+    {
```
Comment (hkaiser) [inline hpx/runtime/threads/detail/scheduling_loop.hpp]: Please align the code formatting to the rest of the file.

File: src/runtime/threads/detail/thread_pool.cpp (hunk 1 above)
Comment (sithhell) [inline src/runtime/threads/detail/thread_pool.cpp]: It's deleted in the dtor of `thread_specific_ptr`.

Comment (hkaiser) [inline src/runtime/threads/detail/thread_pool.cpp]: Nope. See: https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/thread_specific_ptr.hpp#L49-L82

**Pull request merged**